### PR TITLE
chore(test): update testing API

### PR DESCRIPTION
**Pull request by christian-bromann**
This adds some new interfaces including `screenshotTimeout`.

### CODE REVIEW

File: docs/testing/stencil-testrunner/02-config.md
Comment (tanner-reits) [inline docs/testing/stencil-testrunner/02-config.md:19]: Giving a 404 I think cuz of the space in the URL

```suggestion
https://github.com/ionic-team/stencil/blob/d847e92fbc297754cb8dbb7f633de9ce906f54ac/src/declarations/stencil-public-compiler.ts#L1892-L2044
```

**Review by tanner-reits — CHANGES_REQUESTED** [review]

File: docs/testing/stencil-testrunner/02-config.md
Comment (rwaskiewicz) [reply docs/testing/stencil-testrunner/02-config.md:19]: We'd have to update this SHA after we land the feature right? Otherwise, what happens when that commit gets cleaned up by GitHub?

Comment (christian-bromann) [reply docs/testing/stencil-testrunner/02-config.md]: Ups, removed the space. 

> We'd have to update this SHA after we land the feature right?

No need for that. GitHub keeps references to all SHAs around forever. While it might feel weird to use a commit sha that is not part of `main` it should't matter for this use case as every commit remains on GitHub once pushed.

**Review by rwaskiewicz — APPROVED** [review]

**Review by tanner-reits — APPROVED** [review]

**Comment by rwaskiewicz** [comment]
@christian-bromann LGTM! Any reason we can't land this now? I think that 3.14.0 will be our next release anyway, and we already have items for it in `next` here

**Comment by christian-bromann** [comment]
Sounds good, merging then.

**Pull request merged**
